### PR TITLE
[Merged by Bors] - feat: codimension two subspaces have path-connected complement

### PR DESCRIPTION
**Pull request by grunweg**
From sphere-eversion (not written by me).
This basically shows that subspaces of codimension at least two are ample sets (PRed in #11342).

---------

In sphere-eversion, the hypotheses was spelled as "2 \leq Module.rank E"; I have changed this to "1 < Module.rank E" to match the remaining file. (The latter seems to be used more in mathlib, but either is pretty rare.)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Convex/Normed.lean [hunk 1]
```diff
@@ -155,3 +153,87 @@ theorem isConnected_setOf_sameRay_and_ne_zero {x : E} (hx : x ≠ 0) :
...
+theorem joinedIn_compl_zero_of_not_mem_span {x y : E} (hx : x ≠ 0)
+    (hy : y ∉ Submodule.span ℝ {x}) : JoinedIn ({0}ᶜ : Set E) x y := by
+  refine JoinedIn.ofLine lineMap_continuous.continuousOn
+    (lineMap_apply_zero _ _) (lineMap_apply_one _ _) ?_
+  rw [← segment_eq_image_lineMap]
+  exact fun t ht h' ↦ (mt (mem_span_of_zero_mem_segment hx) hy) (h' ▸ ht)
+
+/-- In a real vector space whose dimension is at least two,
+the complement of `{0}` is path-connected. -/
+theorem isPathConnected_compl_zero_of_two_le_dim (hdim : 2 ≤ Module.rank ℝ E) :
```
Comment (sgouezel) [inline Mathlib/Analysis/Convex/Normed.lean:194]: We already have [isPathConnected_compl_singleton_of_one_lt_rank](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/Connected.html#isPathConnected_compl_singleton_of_one_lt_rank)

Comment (grunweg) [reply Mathlib/Analysis/Convex/Normed.lean]: Well spotted; addressed now.

**Comment by grunweg** [comment]
Rebased on master (sorry for the force-push; only the last commit is new since the last review comment).

**Comment by grunweg** [comment]
Bumped heartbeats in `CondexpL1.lean`; I have no idea why my change pushed things over the edge.

**Comment by grunweg** [comment]
Good point; I agree. I have removed these lemmas and moved `isPathConnected_compl_of_two_le_codim` to `Analysis.NormedSpace.Connected` as well. I am not convinced if `Submodule.connectedComponentIn_eq_self_of_two_le_codim` is carrying its weight now; I can certainly remove it now and reconsider in the dependent PRs.
awaiting-review

**Comment by grunweg** [comment]
Wait, let me clean up the diff first.

**Comment by grunweg** [comment]
Now it's good for another look.
awaiting-review

File: Mathlib/Analysis/NormedSpace/Connected.lean [hunk 2]
```diff
@@ -168,3 +168,30 @@ theorem isPreconnected_sphere (h : 1 < Module.rank ℝ E) (x : E) (r : ℝ) :
   · simpa [hr] using isPreconnected_empty
 
 end NormedSpace
+
+section
+
+variable {E F : Type*} [AddCommGroup F] [Module ℝ F] [TopologicalSpace F]
+  [AddCommGroup E] [Module ℝ E] [TopologicalSpace E] [TopologicalAddGroup F] [ContinuousSMul ℝ F]
```
Comment (sgouezel) [inline Mathlib/Analysis/NormedSpace/Connected.lean:175]: Do you really use `E`? Below it seems it's only a submodule.

**Comment by grunweg** [comment]
awaiting-review

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Comment by grunweg** [comment]
Thank you for the reviews!

**Pull request closed without merging**